### PR TITLE
Upgrade to support papertrail version 5

### PR DESCRIPTION
**Pull request by justindoody**
image

I didn't actually need to change any business logic (fixed that nil casting on whodunnit though). Most of this diff is just trivial re-organizations.

### CODE REVIEW

File: fast_versioning.gemspec [hunk 1]
```diff
@@ -3,18 +3,18 @@ $:.push File.expand_path("../lib", __FILE__)
...
   s.description = 'Fast versioning extension for paper_trail'
   s.license     = 'MIT'
 
-  s.files = Dir['{app,config,db,lib}/**/*', 'Rakefile', 'README.md']
+  s.files = Dir['{app,db,lib}/**/*', 'Rakefile', 'README.md']
 
-  s.add_dependency 'rails'
-  s.add_dependency 'paper_trail', '~>4.2'
+  s.add_dependency 'rails', '~> 5.0.7'
+  s.add_dependency 'paper_trail', '~> 5.2'
```
Comment (iiwo) [inline fast_versioning.gemspec]: nothing you did here seems to make this rails 4, or paper_trail 4 incompatible, so maybe we should do:
```
s.add_dependency 'rails', '>= 4.2', '< 6.0'
s.add_dependency 'paper_trail', '>= 4.2', '< 6.0'
```

File: Gemfile.lock [hunk 2]
```diff
@@ -1,105 +1,104 @@
 PATH
```
Comment (iiwo) [inline Gemfile.lock]: we should probably [remove Gemfile.lock from the repo](https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/)

File: lib/fast_versioning/paper_trail_extensions.rb [hunk 3]
```diff
@@ -12,19 +12,24 @@ def create_fast_versions
...
-              meta: tracked_attribute.meta,
-              created_at: created_at
-            )
-          end
+          name = tracked_attribute.name
+          new_value = value_change.value_became(name)
+          prev_value = value_change.value_was(name)
+
+          # Skip if value unchanged.
+          next if prev_value == new_value
```
Comment (iiwo) [inline lib/fast_versioning/paper_trail_extensions.rb:20]: 👍

File: fast_versioning.gemspec (hunk 1 above)
Comment (justindoody) [reply fast_versioning.gemspec]: ah yeah fair enough -- it's all just dependent on any PaperTrail's api changes not rails -- was being extra paranoid but will switch back considering this is public facing

Comment (justindoody) [reply fast_versioning.gemspec]: hmm actually it looks like specs do fail on 5.2

Comment (justindoody) [reply fast_versioning.gemspec]: as this is just a temporary step (will be updating to higher papertrail versions soon i'm going to lock it to < 5.1

File: db/migrate/20160914161314_create_fast_versioning_fast_versions.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-class CreateFastVersioningFastVersions < ActiveRecord::Migration
+class CreateFastVersioningFastVersions < ActiveRecord::Migration[4.2]
```
Comment (iiwo) [inline db/migrate/20160914161314_create_fast_versioning_fast_versions.rb:1]: I think that syntax will actually fail on rails 4, we probably need a `Rails::VERSION::MAJOR` conditional for this like ie. [here](https://github.com/plataformatec/devise/blob/master/lib/generators/active_record/devise_generator.rb#L94)

**Pull request merged**